src/app/report/edit-display/edit-display.component.ts

```diff
@@ -84,9 +88,13 @@ export class EditDisplayComponent {
   showReport(report: Report): void {
     this.disableEditing();
     this.report = report;
-    report.xml
-      ? this.editor.setNewReport(report.xml)
-      : this.editor.setNewReport(this.helperService.convertMessage(report));
+    if (report.xml) {
+      this.editor.setNewReport(report.xml);
+      this.uid = undefined;
```

Comment:
I think it's better to use null here instead of undefined, undefined usually means that a variable was declared but does not have a value yet, while null is an assigned value.
Check [this post](https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript) for some examples.
```suggestion
      this.uid = null;
```

Reply:
I see now that I don't use that variable anymore, because I replaced it with a UpdateReportUtil function, so I removed the whole variable.